main.c

```diff
@@ -274,14 +300,16 @@ multithreaded(struct task_t *task, struct config_t *config)
         pthread_create(&threads[i], NULL, mt_worker, (void *) &context);
     }
 
-    bool found = false;
+    task->from = 0;
```

Comment:
It's easier to iterate suffix in producer and prefix in consumer.
in producer:
task->from = 2
task->to = config->length;
in consumer:
task->to = task->from;
task->from = 0;

Common practice is to treat upper bound as non-inclusive. I.e.
for (i = task->from; i < task->to; ++i)